src/main/java/org/pac4j/j2e/filter/RequiresAuthenticationFilter.java

```diff
@@ -83,255 +91,184 @@ protected final void internalFilter(final HttpServletRequest request, final Http
             final FilterChain chain) throws IOException, ServletException {
 
         final WebContext context = new J2EContext(request, response);
+        final ProfileManager manager = new ProfileManager(context);
```

Comment:
@miremond Sorry Michaël, I chose to simplify your smart logic by decoupling the `CallbackFilter` and the `RequiresAuthenticationFilter` (the source code is slightly different). It should make easier to understand.